numpy/core/src/multiarray/buffer.c

```diff
@@ -833,13 +862,34 @@ gentype_getbuffer(PyObject *self, Py_buffer *view, int flags)
     }
 #endif
     view->len = elsize;
+    if (PyArray_IsScalar(self, Datetime) || PyArray_IsScalar(self, Timedelta)) {
+        elsize = 1; /* descr->elsize is 8, but scalars are always 1 */
+    }
     view->itemsize = elsize;
 
     Py_DECREF(descr);
 
     view->readonly = 1;
     view->suboffsets = NULL;
     view->obj = self;
+    if (view->format) {
+        /* XXX remove me, this should never occur */
```

Comment:
I assume that in the tests this does occur? If so, perhaps say when it occurs in the comment.